packages/mongo/mongo_driver.js

```diff
@@ -345,14 +341,12 @@ MongoConnection.prototype._insert = function (collection_name, document,
   }
 
   var write = self._maybeBeginWrite();
-  var refresh = function () {
-    Meteor.refresh({collection: collection_name, id: document._id });
-  };
-  callback = bindEnvironmentForWrite(writeCallback(write, refresh, callback));
+  callback = bindEnvironmentForWrite(writeCallback(write, callback));
   try {
     var collection = self.rawCollection(collection_name);
     collection.insert(replaceTypes(document, replaceMeteorAtomWithMongo),
                       {safe: true}, callback);
+    Meteor.refresh({collection: collection_name, id: document._id });
```

Comment:
I don't think this is correct, unless I'm missing something.  In the past, we waited until after the database told us that the write had occurred in order to refresh (ie, to wait until oplog is caught up, or do another poll).  Now you're sending out that notification immediately upon sending the command to the database.  So eg for the polling observe driver, you could do a poll immediately and think you've refreshed before the write actually happens.  And for oplog, I guess you're making some assumption about how the arming of the fence relates to the callback being run, which I'm not sure is correct, but since the polling version is definitely incorrect I'm not going to think about this too hard right now.